### ISSUE TITLE
Feature request: Fill only limited number of rows

### ISSUE THREAD

**Issue by estroger34** [tidyverse/tidyr#257]
It would be handy to have an option for the fill function to fill only a limited number of NAs with the previous entry, not the entire NA block. The background is data from measurement instruments recording a value at irregular times which is, however, valid for a limited time period only (e.g. 220 seconds), not until the next timestamp (row) in the data frame.

Currently I do this with a rowwise approach (see below), which is rather slow, obviously. 

```
library(tidyverse)
library(lubridate)

  #generate time series data
  #Note that in reality the interval between two timestamps is irregular, i.e. between 500- 700 s
  data <- bind_cols(
    data_frame(td=seq(from = ymd_hm("2010-01-01 00:00"), 
                                     to = ymd_hm("2010-01-31 23:59"), 
                                     by = 612)),
    as_data_frame(replicate(3,runif(4377))))
        
  #expand with valid measurement periods, then complete
  data2 <- data %>%
    rowwise() %>%
    mutate(temp = list(seq(from = td, to = td + seconds(220), by = 1))) %>%
    unnest() %>%
    select(td = temp, everything(), -td) %>%
    complete(td = full_seq(td,1)) 
```

I have found another approach which might be faster, but not very clean/elegant, either (see http://stackoverflow.com/questions/37972673/fill-a-limited-number-of-values-tidyr-fill).

I would love to be able to tell fill to only fill the next 220 rows after a non-NA entry and leave all others untouched:

```
data2 <- data %>%
    complete(td = full_seq(td,1)) %>%
    fill(V1:V3,nmax = 220)
```

**Comment by hadley** [comment]
I'd say this is out of scope for tidyr sorry. Sounds like you need a more general "rolling" fill.
